test/npu-xrt/add_one_two_runlist/test.cpp

```diff
@@ -0,0 +1,167 @@
+
+int main(int argc, const char *argv[]) {
+  // Program arguments parsing
+  cxxopts::Options options("add_one_two");
+  test_utils::add_default_options(options);
+
+  cxxopts::ParseResult vm;
+  test_utils::parse_options(argc, argv, options, vm);
+
+  std::vector<uint32_t> instr_v =
+      test_utils::load_instr_binary(vm["instr"].as<std::string>());
+
+  int verbosity = vm["verbosity"].as<int>();
+  if (verbosity >= 1)
+    std::cout << "Sequence instr count: " << instr_v.size() << "\n";
+
+  // Start the XRT test code
+  // Get a device handle
+  unsigned int device_index = 0;
+  auto device = xrt::device(device_index);
+
+  // Load the xclbin
+  if (verbosity >= 1)
+    std::cout << "Loading xclbin: " << vm["xclbin"].as<std::string>() << "\n";
+  auto xclbin = xrt::xclbin(vm["xclbin"].as<std::string>());
+
+  // Get the kernel from the xclbin
+  auto xkernels = xclbin.get_kernels();
+  auto xkernel0 = *std::find_if(xkernels.begin(), xkernels.end(),
+                                [](xrt::xclbin::kernel &k) {
+                                  auto name = k.get_name();
+                                  std::cout << "Name: " << name << std::endl;
+                                  return name == "ADDONE";
+                                });
+  auto kernelName0 = xkernel0.get_name();
+  auto xkernel1 = *std::find_if(xkernels.begin(), xkernels.end(),
+                                [](xrt::xclbin::kernel &k) {
+                                  auto name = k.get_name();
+                                  std::cout << "Name: " << name << std::endl;
+                                  return name == "ADDTWO";
+                                });
+  auto kernelName1 = xkernel1.get_name();
+
+  if (verbosity >= 1)
+    std::cout << "Registering xclbin: " << vm["xclbin"].as<std::string>()
+              << "\n";
+
+  device.register_xclbin(xclbin);
+
+  // get a hardware context
+  if (verbosity >= 1)
+    std::cout << "Getting hardware context.\n";
+  xrt::hw_context context(device, xclbin.get_uuid());
+
+  // get a kernel handle
+  if (verbosity >= 1)
+    std::cout << "Getting handle to kernels: " << kernelName0 << " and "
+              << kernelName1 << "\n";
+
+  auto kernel0 = xrt::kernel(context, kernelName0);
+
+  auto bo0_instr = xrt::bo(device, instr_v.size() * sizeof(int),
+                           XCL_BO_FLAGS_CACHEABLE, kernel0.group_id(1));
+  auto bo0_inA = xrt::bo(device, IN_SIZE * sizeof(int32_t),
+                         XRT_BO_FLAGS_HOST_ONLY, kernel0.group_id(3));
+  auto bo0_out = xrt::bo(device, OUT_SIZE * sizeof(int32_t),
+                         XRT_BO_FLAGS_HOST_ONLY, kernel0.group_id(4));
+
+  auto kernel1 = xrt::kernel(context, kernelName1);
+
+  auto bo1_instr = xrt::bo(device, instr_v.size() * sizeof(int),
+                           XCL_BO_FLAGS_CACHEABLE, kernel1.group_id(1));
+  auto bo1_inA = xrt::bo(device, IN_SIZE * sizeof(int32_t),
+                         XRT_BO_FLAGS_HOST_ONLY, kernel1.group_id(3));
```

Comment:
This is likely incorrect since I am passing `bo0_out` to `kernel1.group_id(3)`. @fifield @andrej